File: 1203/1203.cpp
class Solution {
public:
    int sortItemsByGroupsRespectingDep(TreeNode* root) {
        if(!root) return 0;
        return 1+max(sortItemsByGroupsRespectingDep(root->left),sortItemsByGroupsRespectingDep(root->right));
    }
};

File: 1203/1203.go
func sortItemsByGroupsRespectingDep(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := sortItemsByGroupsRespectingDep(root.Left)
	r := sortItemsByGroupsRespectingDep(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}